apps/heft/src/plugins/TypeScriptPlugin/TypeScriptBuilder.ts

```diff
@@ -494,120 +512,133 @@ export class TypeScriptBuilder extends SubprocessRunnerBase<ITypeScriptBuilderCo
-                queueLinkOrCopy({
-                  linkTargetPath: path.join(cacheOutFolderPath, jsMapFilename),
-                  newLinkPath: path.join(outFolderPath, jsMapFilename)
-                });
-              }
-
-              // Write the .js file last in case something is watching its timestamp
-              if (!sourceFile.isDeclarationFile) {
-                const jsFilename: string = `${relativeFilenameWithoutExtension}${jsExtensionOverride}`;
-                queueLinkOrCopy({
-                  linkTargetPath: path.join(cacheOutFolderPath, jsFilename),
-                  newLinkPath: path.join(outFolderPath, jsFilename)
-                });
-              }
-            }
-          }
-        }
+    if (eslint) {
+      eslint.reportFailures();
+    }
+
+    if (tslint) {
+      tslint.reportFailures();
+    }
+
+    if (typeScriptErrorCount > 0) {
+      throw new Error(`Encountered TypeScript error${typeScriptErrorCount > 1 ? 's' : ''}`);
+    }
+  }
+
+  public async _runSolutionBuild(
+    ts: ExtendedTypeScript,
+    measureTsPerformance: PerformanceMeasurer
+  ): Promise<void> {
+    // Ensure the cache folder exists
+    this._cachedFileSystem.ensureFolder(this._configuration.buildCacheFolder);
+
+    this._typescriptTerminal.writeVerboseLine(`Using solution mode`);
+
+    //#region CONFIGURE
+    const { duration: configureDurationMs } = measureTsPerformance('Configure', () => {
+      this._overrideTypeScriptReadJson(ts);
+      const _tsconfig: TTypescript.ParsedCommandLine = this._loadTsconfig(ts);
 
-        await Promise.all(linkPromises);
+      this._validateTsconfig(ts, _tsconfig);
 
-        return { linkCount };
+      EmitFilesPatch.install(ts, _tsconfig, this._moduleKindsToEmit);
+
+      return {};
+    });
+    this._typescriptTerminal.writeVerboseLine(`Configure: ${configureDurationMs}ms`);
+    //#endregion
+
+    const rawDiagnostics: TTypescript.Diagnostic[] = [];
+    const reportDiagnostic: TTypescript.DiagnosticReporter = (diagnostic: TTypescript.Diagnostic) => {
+      rawDiagnostics.push(diagnostic);
+    };
+
+    const solutionBuilderHost: TSolutionHost = this._buildSolutionBuilderHost(ts, reportDiagnostic);
+
+    const solutionBuilder: TTypescript.SolutionBuilder<TTypescript.EmitAndSemanticDiagnosticsBuilderProgram> =
+      ts.createSolutionBuilder(solutionBuilderHost, [this._configuration.tsconfigPath], {});
+
+    const lintPromises: Promise<LinterBase<unknown>>[] = [];
+
+    const [eslintLogger, tslintLogger] = await Promise.all([
+      this._initESlintLogger(),
+      this._initTSlintLogger()
+    ]);
+
+    solutionBuilderHost.afterProgramEmitAndDiagnostics = (
+      program: TTypescript.EmitAndSemanticDiagnosticsBuilderProgram
+    ) => {
+      const tsProgram: TTypescript.Program | undefined = program.getProgram();
+
+      if (tsProgram) {
+        const extendedProgram: IExtendedProgram = tsProgram as IExtendedProgram;
+        if (eslintLogger) {
```

Comment:
It feels a little awkward to running the linters based on whether a logger exists or not. Maybe the logger can get wrapped up into a more generic "linter" object that gets invoked.